### ISSUE TITLE
Ressource Colletion 

### ISSUE THREAD

**Issue by arvenom** [ebizmarts/magento2-mandrill#4]
When trying to compile I get this error: 
Fatal error: Class 'Magento\Framework\Model\Resource\Db\Collection\AbstractCollection'
Changing from Model/Resource to Model/ResourceModel fixes that but after trying to run compiler again it fails with this: 

Ebizmarts\Mandrill\Block\Adminhtml\System\Config\Form\Field\MagentoTemplates
        Incorrect dependency in class Ebizmarts\Mandrill\Block\Adminhtml\System\Config\Form\Field\MagentoTemplates in /Applications/MAMP/htdocs/theil/htdocs/app/code/Ebizmarts/Mandrill/Block/Adminhtml/System/Config/Form/Field/MagentoTemplates.php
\Psr\Log\LoggerInterface already exists in context object
    Ebizmarts\Mandrill\Controller\Adminhtml\Email\Test
        Incompatible argument type: Required type: \Magento\Backend\App\Action\Context. Actual type: \Magento\Framework\App\Action\Context; File:
/Applications/MAMP/htdocs/theil/htdocs/app/code/Ebizmarts/Mandrill/Controller/Adminhtml/Email/Test.php

```
Ebizmarts\Mandrill\Controller\Autoresponder\Unsubscribe
    Incorrect dependency in class Ebizmarts\Mandrill\Controller\Autoresponder\Unsubscribe in /Applications/MAMP/htdocs/theil/htdocs/app/code/Ebizmarts/Mandrill/Controller/Autoresponder/Unsubscribe.php
```

\Magento\Framework\ObjectManagerInterface already exists in context object
    Ebizmarts\Mandrill\Helper\Data
        Incorrect dependency in class Ebizmarts\Mandrill\Helper\Data in /Applications/MAMP/htdocs/theil/htdocs/app/code/Ebizmarts/Mandrill/Helper/Data.php
\Psr\Log\LoggerInterface already exists in context object

**Comment by SantiParagarino** [comment]
Hello @arvenom,

Thanks for letting us know this. I'll try to address this issue today.

**Comment by gonzaloebiz** [comment]
This is fixed in Release 3.0.1

Thanks for your contribution @arvenom
